prov/efa/src/efa_hmem.c

```diff
@@ -320,7 +320,8 @@ static int efa_domain_hmem_info_init_synapseai(struct efa_domain *efa_domain)
 	info->p2p_disabled_by_user = false;
 	/* SynapseAI currently requires P2P */
 	info->p2p_required_by_impl = true;
-	info->p2p_supported_by_device = true;
+	info->p2p_supported_by_device = efa_domain_hmem_support_p2p(efa_domain, FI_HMEM_SYNAPSEAI);
+	info->dmabuf_support_status = EFA_DMABUF_SUPPORTED;
```

Comment:
I think if you have another function for `efa_domain_hmem_support_dmabuf` and make iface as an input parameter, we can get rid of these separate `efa_domain_hmem_info_init_<iface>` functions and just call one function with different ifaces in a loop

Reply:
Good point. I prefer to do that separately. It needs some nice print for different iface. I need to change common code for that.

Reply:
I figured there is `fi_tostr` util. Refactored this part in update.